yarn-project/end-to-end/src/uniswap_trade_on_l1_from_l2.test.ts

```diff
@@ -407,5 +427,316 @@ describe('uniswap_trade_on_l1_from_l2', () => {
     logger('***** 🧚‍♀️ SWAP L2 assets on L1 Uniswap 🧚‍♀️ *****');
     logger('WETH balance after swap : ', wethL2BalanceAfterSwap.toString());
     logger('DAI balance after swap  : ', daiL2BalanceAfterSwap.toString());
-  }, 140_000);
+  });
+
+  // Edge cases for the private flow:
+  // note - tests for uniswapPortal.sol and minting asset on L2 are covered in other tests.
+
+  it("uniswap contract can't unshield if I don't give the right approval", async () => {
```

Comment:
Would be more specific with the "right approval", e.g.., 
```suggestion
  it("swap_private reverts without unshield approval", async () => {
```